libethdrivers/include/ethdrivers/raw.h

```diff
@@ -156,4 +156,11 @@ struct dma_buf_cookie {
     void* pbuf;
 };
 
+struct eth_plat_config {
+    void* buffer_addr;
+    uint8_t mac_addr[6];
+    uint8_t prom_mode;
+    ethif_driver_init init;
```

Comment:
As far a I can tell this member is never referenced, so I cannot divine it's intended purpose

Reply:
This is used to make an platform independent driver. Instead of doing 

```#ifdef CONFIG_PLAT_IMX6
ethif_imx6_init
#elif CONFIG_PLAT_ZYNQ7000
ethif_zynq7000_init
#endif
```
So the Ethdriver component now has an platform init function call declared locally that sets the init member, so the function call is 
`    error = config->init(&eth_driver, ioops, (void *)config);`

Reply:
For reference, here is how we implemented an Ethdriver that functions with the zynq7000 and the imx6.

https://github.com/dornerworks/camkes-debug-tool/blob/master/components/Ethdriver/src/ethdriver.c

Reply:
My objection here is that the `init` function here is not a configuration to the driver, it's a configuration in your component. I would suggest having your own config struct in your component like

```c
struct ethdriver_config {
    eth_plat_config driver_config;
    ethif_driver_init driver_init;
};
```

Reply:
Understood. We will remove the component specific `init`  from the driver structure.